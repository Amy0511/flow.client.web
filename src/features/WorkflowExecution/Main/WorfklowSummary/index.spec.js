import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import WorkflowSummary from "./index";

const workflowData = {
  name: "test",
  shortDescription: "test"
};

const workflowExecutionData = {
  status: "",
  creationDate: "",
  duration: ""
};

const version = 1;

describe("WorkflowSummary --- Snapshot", () => {
  it("Capturing Snapshot of WorkflowSummary", () => {
    const renderedValue = renderer.create(
      <WorkflowSummary workflowExecutionData={workflowExecutionData} workflowData={workflowData} version={version} />
    );
    expect(renderedValue).toMatchSnapshot();
  });
});

describe("WorkflowSummary --- Shallow render", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <WorkflowSummary workflowExecutionData={workflowExecutionData} workflowData={workflowData} version={version} />
    );
  });

  it("Render the DUMB component", () => {
    expect(wrapper.length).toEqual(1);
  });
});
