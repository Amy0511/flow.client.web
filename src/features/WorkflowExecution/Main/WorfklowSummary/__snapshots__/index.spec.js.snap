// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkflowSummary --- Snapshot Capturing Snapshot of WorkflowSummary 1`] = `
<div
  className="c-workflow-summary"
>
  <div
    className="s-workflow-summary-title"
  >
    test
  </div>
  <div
    className="c-workflow-summary-details"
  >
    <div
      className="b-workflow-summary-field"
    >
      <div
        className="b-workflow-summary-field__key"
      >
        Description
      </div>
      <div
        className="b-workflow-summary-field__value"
      >
        test
      </div>
    </div>
    <div
      className="b-workflow-summary-field"
    >
      <div
        className="b-workflow-summary-field__key"
      >
        Version
      </div>
      <div
        className="b-workflow-summary-field__value"
      >
        1
      </div>
    </div>
    <div
      className="b-workflow-summary-field"
    >
      <div
        className="b-workflow-summary-field__key"
      >
        Status
      </div>
      <div
        className="b-workflow-summary-field__value"
      >
        Not Started
        <img
          alt="Status "
          className="b-activity-card__status-icon"
          src="pending.svg"
        />
      </div>
    </div>
    <div
      className="b-workflow-summary-field"
    >
      <div
        className="b-workflow-summary-field__key"
      >
        Start Time
      </div>
      <div
        className="b-workflow-summary-field__value"
      >
        Invalid date
      </div>
    </div>
    <div
      className="b-workflow-summary-field"
    >
      <div
        className="b-workflow-summary-field__key"
      >
        Duration
      </div>
      <div
        className="b-workflow-summary-field__value"
      >
        ---
      </div>
    </div>
  </div>
</div>
`;
